### PR TITLE
Chore: Give option to use existing p directory

### PR DESCRIPTION
**Pull request by humbletree**
While coding I didn't realize that in Go it is best practice to use tabs instead of spaces. Majority of file had tabs, at few places it had spaces. While fixing the indentation of my changes, fixed indentation at other places as well.

---
This change is [Reviewable](https://reviewable.io/reviews/dgraph-io/dgraph/7735)

### CODE REVIEW

File: compose/compose.go [hunk 1]
```diff
@@ -572,6 +575,85 @@ func fatal(err error) {
...
+	}
+	return os.Chmod(dst, srcinfo.Mode())
+}
+
+func copyDir(src string, dst string) error {
+	var err error
+	var fds []os.FileInfo
+	var srcinfo os.FileInfo
+
+	if srcinfo, err = os.Stat(src); err != nil {
```
Comment (roshbhatia) [inline compose/compose.go:626]: Nitpick: when naming variables, functions, etc., it's best to stick to what go calls [mixedCaps](https://golang.org/doc/effective_go#mixed-caps) to stay idiomatic-ish.

Comment (roshbhatia) [reply compose/compose.go:626]: I recommend looking at the go std libraries for a good reference on golang style, that what I typically refer to. For ex: [strings](https://github.com/golang/go/blob/master/src/strings/strings.go)

**Review by roshbhatia — APPROVED** [review]

File: compose/compose.go (hunk 1 above)
Comment (roshbhatia) [reply compose/compose.go:626]: Also for the formatting issues you mentioned in this PR, check out `go fmt`

File: compose/compose.go [hunk 2]
```diff
@@ -572,6 +575,85 @@ func fatal(err error) {
...
+		return err
+	}
+
+	if err = os.MkdirAll(dst, srcinfo.Mode()); err != nil {
+		fatal(errors.Errorf("Error in making dir %v. Error: %v", dst, err))
+		return err
+	}
+
+	if fds, err = ioutil.ReadDir(src); err != nil {
+		fatal(errors.Errorf("Error in reading source dir %v. Error: %v", src, err))
```
Comment (roshbhatia) [inline compose/compose.go:637]: Looks like the error logs in this PR is a ~little inconsistent, for ex: compare this thrown error, with [this one](https://github.com/dgraph-io/dgraph/blob/13ace774970835ddea4a0cc8a6e9728fae749103/compose/compose.go#L646) -- the format is a little different in terms of punctuation/capitalization. One option is, to keep it consistent, format the error string along the lines of [this one](https://github.com/dgraph-io/dgraph/blob/69c83890a4511677a17dcb53935fcedf12c8a875/compose/compose.go#L779)? Wdyt?

**Review by roshbhatia — CHANGES_REQUESTED** [review]

File: compose/compose.go [hunk 3]
```diff
@@ -572,6 +575,85 @@ func fatal(err error) {
...
+
+func copyFile(src, dst string) error {
+	var err error
+	var srcfd *os.File
+	var dstfd *os.File
+	var srcinfo os.FileInfo
+
+	if srcfd, err = os.Open(src); err != nil {
+		fatal(errors.Errorf("Error in opening source file %v. Error: %v", src, err))
+		return err
```
Comment (roshbhatia) [inline compose/compose.go:599]: Will this function return an error? Looks like it calls `fatal`, which will exit the program, right? IMO, best to just return the error and have it handled by the code that is calling copyFile

File: compose/compose.go [hunk 4]
```diff
@@ -572,6 +575,85 @@ func fatal(err error) {
...
+	var fds []os.FileInfo
+	var srcinfo os.FileInfo
+
+	if srcinfo, err = os.Stat(src); err != nil {
+		fatal(errors.Errorf("Error in doing stat of source dir %v. Error: %v", src, err))
+		return err
+	}
+
+	if err = os.MkdirAll(dst, srcinfo.Mode()); err != nil {
+		fatal(errors.Errorf("Error in making dir %v. Error: %v", dst, err))
```
Comment (roshbhatia) [inline compose/compose.go:632]: see above comment about the copyFile function, applies here

File: compose/compose.go [hunk 5]
```diff
@@ -752,6 +844,26 @@ func main() {
...
 	}
 
+	if opts.PDir != "" {
+		if _, err := os.Stat(opts.DataDir); !os.IsNotExist(err) {
+			fatal(errors.Errorf("Directory %v already exists.", opts.DataDir))
+		}
+
+		n := 1
+		for n <= opts.NumAlphas {
+			new_dir := opts.DataDir + "/alpha" + strconv.Itoa(n) + "/p"
```
Comment (roshbhatia) [inline compose/compose.go:854]: nitpick: `new_dir` doesn't conform to mixedCaps or the naming convention in the above nitpick -- best to keep the style consistent

**Review by roshbhatia — APPROVED** [review]
lgtm!

**Review by danielmai — CHANGES_REQUESTED** [review]
Address the comments and then :lgtm: 

> Reviewed 1 of 1 files at r3.
> *[Reviewable](https://reviewable.io/reviews/dgraph-io/dgraph/7735#-:-M_aNEkJE4cMc3sDMKmv:bsjsp52)* status: all files reviewed, 6 unresolved discussions (waiting on @humbletree, @manishrjain, and @vvbalaji-dgraph)

___
*[compose/compose.go, line 665 at r3](https://reviewable.io/reviews/dgraph-io/dgraph/7735#-M_aLzAcEBabn8Qp5T5x:-M_aLzAcEBabn8Qp5T5y:b4xamxp) ([raw file](https://github.com/dgraph-io/dgraph/blob/366557e854e33e774db3e43da70ffbdcaea0d7fa/compose/compose.go#L665)):*
> ```Go
> 	cmd.PersistentFlags().StringVarP(&opts.DataDir, "data_dir", "d", "",
> 		"mount a host directory as /data in containers")
> 	cmd.PersistentFlags().StringVarP(&opts.PDir, "p_dir", "p", "",
> ```

Change the long flag from `p_dir` to `postings`, like it is in Alpha. The shorthand can remain `p`.

**Review by danielmai — COMMENTED** [review]
Add some output about running the `/assign` endpoint commands.

> *[Reviewable](https://reviewable.io/reviews/dgraph-io/dgraph/7735#-:-M_aRaxiB0Whncrns5lM:b-g4vxu5)* status: all files reviewed, 6 unresolved discussions (waiting on @humbletree, @manishrjain, and @vvbalaji-dgraph)

File: compose/compose.go (hunk 1 above)
Comment (humbletree) [reply compose/compose.go]: Moved variable to mixed caps. Yes, later to fix formatting I used `gofmt -s compose.go`.

File: compose/compose.go (hunk 2 above)
Comment (humbletree) [reply compose/compose.go]: Made error logging consistent.

File: compose/compose.go (hunk 3 above)
Comment (humbletree) [reply compose/compose.go]: Makes sense. Made function return an error.

File: compose/compose.go (hunk 5 above)
Comment (humbletree) [reply compose/compose.go]: Done.

**Review by humbletree — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/dgraph-io/dgraph/7735)* status: 0 of 1 files reviewed, 6 unresolved discussions (waiting on @danielmai, @manishrjain, @roshbhatia, and @vvbalaji-dgraph)

___
*[compose/compose.go, line 599 at r2](https://reviewable.io/reviews/dgraph-io/dgraph/7735#-MYPZzVDuCBKcqVMxJ4d-r2-599:-M_amGd71eJBCKECmKQr:b-896fix) ([raw file](https://github.com/dgraph-io/dgraph/blob/13ace774970835ddea4a0cc8a6e9728fae749103/compose/compose.go#L599)):*
<details><summary><i>Previously, humbletree (Ashish Gupta) wrote…</i></summary><blockquote>

Makes sense. Made function return an error.
</blockquote></details>

Done.
___
*[compose/compose.go, line 626 at r2](https://reviewable.io/reviews/dgraph-io/dgraph/7735#-MYPZzVDuCBKcqVMxJ4d-r2-626:-M_amHXD41bWG8QdyxyS:b-896fix) ([raw file](https://github.com/dgraph-io/dgraph/blob/13ace774970835ddea4a0cc8a6e9728fae749103/compose/compose.go#L626)):*
<details><summary><i>Previously, humbletree (Ashish Gupta) wrote…</i></summary><blockquote>

Moved variable to mixed caps. Yes, later to fix formatting I used `gofmt -s compose.go`.
</blockquote></details>

Done.
___
*[compose/compose.go, line 632 at r2](https://reviewable.io/reviews/dgraph-io/dgraph/7735#-MYPZzVDuCBKcqVMxJ4d-r2-632:-M_am8YQ7LKPsTvx-qr1:b-896fix) ([raw file](https://github.com/dgraph-io/dgraph/blob/13ace774970835ddea4a0cc8a6e9728fae749103/compose/compose.go#L632)):*
<details><summary><i>Previously, roshbhatia (Roshan Bhatia) wrote…</i></summary><blockquote>

see above comment about the copyFile function, applies here
</blockquote></details>

Done.
___
*[compose/compose.go, line 637 at r2](https://reviewable.io/reviews/dgraph-io/dgraph/7735#-MYPZzVDuCBKcqVMxJ4d-r2-637:-M_am9czEYY7P8YTDr9E:b-896fix) ([raw file](https://github.com/dgraph-io/dgraph/blob/13ace774970835ddea4a0cc8a6e9728fae749103/compose/compose.go#L637)):*
<details><summary><i>Previously, humbletree (Ashish Gupta) wrote…</i></summary><blockquote>

Made error logging consistent.
</blockquote></details>

Done.
___
*[compose/compose.go, line 854 at r2](https://reviewable.io/reviews/dgraph-io/dgraph/7735#-MYPZzVDuCBKcqVMxJ4d-r2-854:-M_amCsP4Yz5Z-t0m2oM:b-896fix) ([raw file](https://github.com/dgraph-io/dgraph/blob/13ace774970835ddea4a0cc8a6e9728fae749103/compose/compose.go#L854)):*
<details><summary><i>Previously, humbletree (Ashish Gupta) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Done.
___
*[compose/compose.go, line 665 at r3](https://reviewable.io/reviews/dgraph-io/dgraph/7735#-M_aLzAcEBabn8Qp5T5x:-M_amBs-BFEdJEGpI3sQ:b-896fix) ([raw file](https://github.com/dgraph-io/dgraph/blob/366557e854e33e774db3e43da70ffbdcaea0d7fa/compose/compose.go#L665)):*
<details><summary><i>Previously, danielmai (Daniel Mai) wrote…</i></summary><blockquote>

Change the long flag from `p_dir` to `postings`, like it is in Alpha. The shorthand can remain `p`.
</blockquote></details>

Done.

**Review by humbletree — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/dgraph-io/dgraph/7735)* status: 0 of 1 files reviewed, 6 unresolved discussions (waiting on @danielmai, @manishrjain, @roshbhatia, and @vvbalaji-dgraph)

___
*[compose/compose.go, line 599 at r2](https://reviewable.io/reviews/dgraph-io/dgraph/7735#-MYPZzVDuCBKcqVMxJ4d-r2-599:-M_amZ_XCKskEJb0luk9:b-896fix) ([raw file](https://github.com/dgraph-io/dgraph/blob/13ace774970835ddea4a0cc8a6e9728fae749103/compose/compose.go#L599)):*
<details><summary><i>Previously, humbletree (Ashish Gupta) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Done.
___
*[compose/compose.go, line 626 at r2](https://reviewable.io/reviews/dgraph-io/dgraph/7735#-MYPZzVDuCBKcqVMxJ4d-r2-626:-M_am_ea-Y4F68Nrg-Js:b-896fix) ([raw file](https://github.com/dgraph-io/dgraph/blob/13ace774970835ddea4a0cc8a6e9728fae749103/compose/compose.go#L626)):*
<details><summary><i>Previously, humbletree (Ashish Gupta) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Done.
___
*[compose/compose.go, line 632 at r2](https://reviewable.io/reviews/dgraph-io/dgraph/7735#-MYPZzVDuCBKcqVMxJ4d-r2-632:-M_amaYu7Ymd6M9ExtGo:b-896fix) ([raw file](https://github.com/dgraph-io/dgraph/blob/13ace774970835ddea4a0cc8a6e9728fae749103/compose/compose.go#L632)):*
<details><summary><i>Previously, humbletree (Ashish Gupta) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Done.

**Pull request merged**